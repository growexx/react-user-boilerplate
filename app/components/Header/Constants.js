/*
 * Header Constants
 * This file contains constants used in Header component.
 */
import React from 'react';
import { UserOutlined, LockOutlined } from '@ant-design/icons';
import { ROUTES } from 'containers/constants';

export const MenuItems = [
  {
    to: ROUTES.PROFILE,
    tabName: 'Profile',
    icon: <UserOutlined />,
  },
  {
    to: ROUTES.CHANGE_PASSWORD,
    tabName: 'Change Password',
    icon: <UserOutlined />,
  },
  {
    to: ROUTES.LOGOUT,
    tabName: 'Logout',
    icon: <LockOutlined />,
  },
];
