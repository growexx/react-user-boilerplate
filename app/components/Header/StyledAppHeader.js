import styled from 'styled-components';

export const StyledAppHeader = styled.div`
  display: flex;
  flex: 1;
  line-height: 0;
`;

export const StyledAppHeaderColored = styled.div`
  display: flex;
  flex: 1;
  line-height: 0;
  background-color: #190426;
  color: white !important;
`;

export const AvatarWrapper = styled.div`
  margin-left: auto;
  margin-right: 16px;
  align-self: center;
  cursor: pointer;
  display: flex;
  align-items: center;
  .anticon-shopping-cart {
    font-size: 18px;
  }
`;
