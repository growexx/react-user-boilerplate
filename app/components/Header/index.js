/* eslint-disable react/no-array-index-key */
/**
 * Avatar/index.js
 *
 * This is the Avatar Component File.
 */
import React from 'react';
import Notification from 'components/Notification';
import Cart from 'components/Cart';
import PropTypes from 'prop-types';
import { MenuItems } from './constants';
import {
  StyledAppHeader,
  AvatarWrapper,
  StyledAppHeaderColored,
} from './StyledAppHeader';
import Avatar from '../Avatar';

const Header = props =>
  props.menuBackground ? (
    <StyledAppHeaderColored {...props}>
      <AvatarWrapper>
        <Cart />
        <Notification />
        <Avatar menu={MenuItems} />
      </AvatarWrapper>
    </StyledAppHeaderColored>
  ) : (
    <StyledAppHeader {...props}>
      <AvatarWrapper>
        <Cart />
        <Notification />
        <Avatar menu={MenuItems} />
      </AvatarWrapper>
    </StyledAppHeader>
  );

Header.propTypes = {
  menuBackground: PropTypes.bool,
};
export default Header;
