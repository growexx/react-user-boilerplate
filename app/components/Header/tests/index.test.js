import React from 'react';
import { render } from 'react-testing-library';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import { ConnectedRouter } from 'connected-react-router/immutable';
import { createMemoryHistory } from 'history';
import { PersistGate } from 'redux-persist/integration/react';

import Header from '../index';
import { StyledAppHeader } from '../StyledAppHeader';
import configureStore from '../../../configureStore';

describe('<Header />', () => {
  const history = createMemoryHistory();
  // const store = configureStore({}, history);
  const initialState = {};
  const { store, persistor } = configureStore(initialState, history);

  it('should render a div', () => {
    const { container } = render(
      <Provider store={store}>
        <PersistGate persistor={persistor}>
          <IntlProvider locale="en">
            <ConnectedRouter history={history}>
              <Header />
            </ConnectedRouter>
          </IntlProvider>
        </PersistGate>
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a div with props', () => {
    render(
      <Provider store={store}>
        <PersistGate persistor={persistor}>
          <IntlProvider locale="en">
            <ConnectedRouter history={history}>
              <Header menuBackground />
            </ConnectedRouter>
          </IntlProvider>
        </PersistGate>
      </Provider>,
    );
  });

  it('should render a div with props', () => {
    const { container } = render(
      <Provider store={store}>
        <PersistGate persistor={persistor}>
          <IntlProvider locale="en">
            <ConnectedRouter history={history}>
              <StyledAppHeader menuBackground />
            </ConnectedRouter>
          </IntlProvider>
        </PersistGate>
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a div with props', () => {
    const { container } = render(
      <Provider store={store}>
        <PersistGate persistor={persistor}>
          <IntlProvider locale="en">
            <ConnectedRouter history={history}>
              <StyledAppHeader />
            </ConnectedRouter>
          </IntlProvider>
        </PersistGate>
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
