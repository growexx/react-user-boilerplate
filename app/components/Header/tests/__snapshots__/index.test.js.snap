// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render a div 1`] = `
.c2 {
  margin-right: 10px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  line-height: 0;
}

.c1 {
  margin-left: auto;
  margin-right: 16px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 .anticon-shopping-cart {
  font-size: 18px;
}

.c3 .ant-btn-group > .ant-btn:first-child {
  display: none;
}

.c3 .ant-dropdown-trigger,
.c3 .ant-dropdown-trigger:hover {
  border-radius: 50% !important;
  background-color: #4d186e;
  color: #fff;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="u-mr-1 u-d-inline-block"
    >
      <button
        class="ant-btn ant-btn-text btn-hover-none p-4"
        data-testid="badge-Cart"
        type="button"
      >
        <span
          class="ant-badge"
        >
          <span
            aria-label="shopping-cart"
            class="anticon anticon-shopping-cart u-font-size-xlg"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="shopping-cart"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 1024 1024"
              width="1em"
            >
              <path
                d="M922.9 701.9H327.4l29.9-60.9 496.8-.9c16.8 0 31.2-12 34.2-28.6l68.8-385.1c1.8-10.1-.9-20.5-7.5-28.4a34.99 34.99 0 00-26.6-12.5l-632-2.1-5.4-25.4c-3.4-16.2-18-28-34.6-28H96.5a35.3 35.3 0 100 70.6h125.9L246 312.8l58.1 281.3-74.8 122.1a34.96 34.96 0 00-3 36.8c6 11.9 18.1 19.4 31.5 19.4h62.8a102.43 102.43 0 00-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7h161.1a102.43 102.43 0 00-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7H923c19.4 0 35.3-15.8 35.3-35.3a35.42 35.42 0 00-35.4-35.2zM305.7 253l575.8 1.9-56.4 315.8-452.3.8L305.7 253zm96.9 612.7c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 01-31.6 31.6zm325.1 0c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 01-31.6 31.6z"
              />
            </svg>
          </span>
        </span>
      </button>
    </div>
    <div
      data-testid="badge-cart-drawer"
      data-visible="false"
    />
    <div
      class="c2"
    >
      <div
        class="u-mr-1 u-d-inline-block"
      >
        <button
          class="ant-btn ant-btn-text btn-hover-none p-4"
          data-testid="badge-Cart"
          type="button"
        >
          <span
            class="ant-badge "
          >
            <span
              aria-label="bell"
              class="anticon anticon-bell u-font-size-xlg"
              data-testid="BELL_ICON"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="bell"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M816 768h-24V428c0-141.1-104.3-257.7-240-277.1V112c0-22.1-17.9-40-40-40s-40 17.9-40 40v38.9c-135.7 19.4-240 136-240 277.1v340h-24c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h216c0 61.8 50.2 112 112 112s112-50.2 112-112h216c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM512 888c-26.5 0-48-21.5-48-48h96c0 26.5-21.5 48-48 48zM304 768V428c0-55.6 21.6-107.8 60.9-147.1S456.4 220 512 220c55.6 0 107.8 21.6 147.1 60.9S720 372.4 720 428v340H304z"
                />
              </svg>
            </span>
          </span>
        </button>
      </div>
    </div>
    <div
      class="c3"
    >
      <div
        class="ant-btn-group ant-dropdown-button"
      >
        <button
          class="ant-btn ant-btn-default"
          type="button"
        />
        <button
          class="ant-btn ant-btn-default ant-btn-icon-only ant-dropdown-trigger"
          type="button"
        >
          <span
            aria-label="user"
            class="anticon anticon-user"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="user"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Header /> should render a div with props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  line-height: 0;
}

<div
  class="c0"
/>
`;

exports[`<Header /> should render a div with props 2`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  line-height: 0;
}

<div
  class="c0"
/>
`;
